tgui/packages/tgui/interfaces/NtosMessenger.tsx

```diff
@@ -0,0 +1,469 @@
+            <Button.Confirm
+              icon="trash-can"
+              content="Delete chat"
+              onClick={() => act('PDA_clearMessages', { ref: recp.ref })}
+            />
+          </Section>
+        </Stack.Item>
+
+        <Stack.Item grow={1}>
+          <Section
+            scrollable
+            fill
+            fitted
+            title={`${recp.name} (${recp.job})`}
+            scrollableRef={this.scrollRef}>
+            <Stack vertical fill className="NtosMessenger__ChatLog">
+              <Stack.Item textAlign="center" fontSize={1}>
+                This is the beginning of your chat with {recp.name}.
+              </Stack.Item>
+              <Stack.Divider />
+              {filteredMessages}
+            </Stack>
+          </Section>
+        </Stack.Item>
+
+        <Stack.Item>
+          <Section>
+            <Stack fill>
+              <Stack.Item grow={1}>
+                <Input
+                  placeholder={`Send message to ${recp.name}...`}
+                  fluid
+                  autofocus
+                  justify
+                  id="input"
+                  value={msg}
+                  maxLength={1024}
+                  onInput={this.handleMessageInput}
+                  onEnter={this.handleSendMessage}
+                />
+              </Stack.Item>
+              <Stack.Item>
+                <Button
+                  icon="arrow-right"
+                  onClick={this.handleSendMessage}
+                  disabled={!canSend}
+                />
+              </Stack.Item>
+            </Stack>
+          </Section>
+        </Stack.Item>
+      </Stack>
+    );
+  }
+}
+
+const SendToAllModal = (_: any, context: any) => {
+  const { data, act } = useBackend<NtosMessengerData>(context);
+  const { on_spam_cooldown } = data;
+
+  const [msg, setMsg] = useLocalState<string>(context, 'everyoneMessage', '');
+
+  return (
+    <>
+      <Section>
+        <Stack justify="space-between">
+          <Stack.Item align="center">
+            <Icon name="satellite-dish" mr={1} />
+            Send To All
+          </Stack.Item>
+          <Stack.Item>
+            <Button
+              icon="arrow-right"
+              disabled={on_spam_cooldown || msg === ''}
+              tooltip={
+                on_spam_cooldown
+                  ? 'Wait before sending more messages!'
+                  : undefined
+              }
+              tooltipPosition="auto-start"
+              onClick={() => {
+                act('PDA_sendEveryone', { msg: msg });
+                setMsg('');
+              }}>
+              Send
+            </Button>
+          </Stack.Item>
+        </Stack>
+      </Section>
+      <Section>
+        <TextArea
+          height={5}
+          value={msg}
+          placeholder="Send message to everyone..."
+          onInput={(_: any, v: string) => setMsg(v)}
+        />
+      </Section>
+    </>
+  );
+};
```

Comment:
i should note that 500 lines is the magic number where you should break this into multiple files. you're almost there, please consider it

Reply:
I think it won't get that far, especially since this new refactor is going to move some code from tgui to byond